enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while to_a.length > i
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    to_a.my_each { |item| new_array.push(item) if yield(item) }
+    new_array
+  end
+
```

Comment:
1. enumerables #my_select returns an enumerator if no block is given

Reply:
Solved the Requested change